h2o-k8s/README.md

```diff
@@ -0,0 +1,103 @@
+# H2O Kubernetes integration
+
+The integration of Kubernetes and H2O is possible via `water.k8s.KubernetesEmbeddedConfigProvider` - to be found
+in this module. This implementation of `EmbeddedConfigProvider` is dynamically loaded on H2O start and remains inactive
+unless H2O is running in a Docker container managed by Kubernetes is detected.  
+
+## Running H2O in K8s - user's guide
+
+H2O Pods deployed on Kubernetes cluster require a 
+[headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) 
+for H2O Node discovery. The headless service, instead of load-balancing incoming requests to the underlying
+H2O pods, returns a set of adresses of all the underlying pods. It is therefore the responsibility of the K8S
+cluster administrator to set-up the service correctly to cover H2O nodes only.
+
+### Creating the headless service
+First, a headless service must be created on Kubernetes.  
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: h2o-service
+spec:
+  type: ClusterIP
+  clusterIP: None
+  selector:
+    app: h2o-k8s
+  ports:
+  - protocol: TCP
+    port: 54321
+```
+
+The `clusterIP: None` defines the service as headless. The `port: 54321` is the default H2O port. Users and client libraries
+use this port to talk to the H2O cluster.
+
+The `app: h2o-k8s` setting is of **great importance**, as this is the name of the application with H2O pods inside. 
+Please make sure this setting corresponds to the name of H2O deployment name chosen.
+
+### Creating the H2O deployment
+
+It is **strongly recommended** to run H2O as a [Stateful set](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/)
+on a Kubernetes cluster. Kubernetes assumes all the pods inside the cluster are stateful and does not attempt to restart
+the individual pods on failure. Once a job is triggered on an H2O cluster, the cluster is locked and no additional nodes
+can be added. Therefore, the cluster has to be restarted as a whole if required - which is a perfect fit for a StatefulSet.
+ 
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: h2o-stateful-set
+  namespace: h2o-statefulset
+spec:
+  serviceName: h2o-service
+  replicas: 3
+  selector:
+    matchLabels:
+      app: h2o-k8s
+  template:
+    metadata:
+      labels:
+        app: h2o-k8s
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: h2o-k8s
+          image: 'pscheidl/h2o-k8s'
+          ports:
+            - containerPort: 54321
+              protocol: TCP
+          env:
```

Comment:
nit: may want to define some sample resources here. For cpu/memory.

Reply:
This is highly individual. I'll mention the need to adjust these settings as well, but will add them.

Reply:
since the general recommendation is min 4GB per node I would stick to that here

Reply:
that is a good suggestion

Reply:
Done. The request is now 4 GB, not the limit (limits, if not strictly enforced, are managed by K8S implementation). We'd only like to request 4 GB per memory. The rests is for the K8S admin to figure out,.